crates/storage/store.rs

```diff
@@ -1365,6 +1414,41 @@ impl Store {
         Ok(store)
     }
 
+    /// Shutdown the store and wait for background threads to finish.
+    /// This should be called before dropping the Store to ensure all background
+    /// threads have completed their work and the backend can be safely closed.
+    ///
+    /// This method consumes the Store. After calling shutdown, the Store should not be used.
+    pub fn shutdown(self) {
+        // Take the thread handles out of the Mutex before dropping self
+        let handles = {
+            let mut threads = self.background_threads.lock().unwrap();
```

Comment:
We can avoid the `unwrap` here:

```suggestion
            let mut threads = self.background_threads.lock().unwrap_or_else(|e| e.into_inner());
```